src/WebJobs.Extensions.DurableTask/ContextInterfaces/IDurableOrchestrationClient.cs

```diff
@@ -141,8 +141,7 @@ public interface IDurableOrchestrationClient
         /// </exception>
         Task<string> StartNewAsync<T>(
             string orchestratorFunctionName,
-            T input)
-            where T : class;
+            T input);
```

Comment:
Hmm, that seems to be [this](https://github.com/Azure/azure-functions-durable-extension/pull/1422) PR. The PR itself is to allow for mocking of APIs during testing. Unfortunately, the `T : class` constraint is added in the first commit and not discussed explicitly on GitHub, so I'm not immediately sure why it was added.

I'll stare at the PR a bit longer, and see if there's any trace of a discussion about this internally;

Reply:
The PR you're looking at only moved this method signature from one file to another. You'll have to go further back in time, looking at when this method signature was introduced originally in DurableContextExtensions.cs.

Reply:
No luck on my end.
Hmm, can you try to remember more details about this is here in the first place?
If we want to be safe, I would also be ok stashing this for now and adding it to the list of potential breaking changes for an eventual new major version.

Reply:
Ah, you're right - GitHub hid the deletion of content in `DurableContextExtensions.cs` from me.
This seems to be real the PR that originates this constraint - https://github.com/Azure/azure-functions-durable-extension/pull/986, and then again on V2 here - https://github.com/Azure/azure-functions-durable-extension/pull/991

**Context**
Brandon describes this very case in the PR itself. In short - this constraint was introduced to allow an orchestration instanceID to be specified without that string being treated as the orchestration input itself.

Before that PR, if you had `startNew("MyOrch", "myInstanceId")` then "myInstanceId" would become the orchestrator input instead of the instanceID due to an overload of signature `startNew(name: string,  input: object)` triggering.

So the fix was to introduce a generic overload and have that generic overload become what is **required** when a user wants to specify a string input but not the instanceID. So now, when a user wants to specify a string input but no instanceId, they need the "generic brackets <>` and do `startNew<string>("myOrch", "myInput")`. and the standard `startNew` without "generic brackets` always treats a second string parameter as the instanceId.

**However:**
I'm not immediately convinced that the `T : class` was required to begin with. I think it should suffice just to create the generic-type overload. I'll make a few local tests on my end to validate. In the meantime, I'd be curious to know if @brandonh-msft remembers why that was added :)

Reply:
Yup the reason was so that, in the absence of the parameter, we could pass `null` (can't do that unless you type-constrain the generic to a `class`). BUT, in the new land of `<Nullable>` we can/should probably change this to just `T?` and remove the constraint.

Reply:
Thanks! That's a good idea. Enabling nullable analysis in this whole file (I understand the team is trying to do file-wide analyses when we enable nullable-checks) might take a bit of time, potentially making this a larger change. But I'll take a note about trying that out next :) .

Reply:
yeah, converting to nullable usually has cascading effects - understood. Nevertheless that's why the type constraint was there :) Alternatively, I suppose, we could use `default` instead of `null`, perhaps?